### ISSUE TITLE
Uncaught ReferenceError: monaco is not defined at new t.MonacoWorkspace

### ISSUE THREAD

**Issue by MassMessage** [TypeFox/monaco-languageclient#285]
I just build the sample in the [example ](https://github.com/TypeFox/monaco-languageclient/tree/master/example)folder (with some modifications to make monaco edit typescript rather json(I'm getting to know how to use monaco api), below in the post) but I got some erros on the console window like below and the autcompleter isn't working, only the syntax highlight.

The error is:

![image](https://user-images.githubusercontent.com/11813549/130395179-9d5310c4-bfc9-4659-a435-195c2d1636e8.png)

the modified client and server:

**client.ts**
```

import { listen, MessageConnection } from 'vscode-ws-jsonrpc';
import * as monaco from 'monaco-editor'
import {
    MonacoLanguageClient, CloseAction, ErrorAction,
    MonacoServices, createConnection
} from 'monaco-languageclient';
import normalizeUrl = require('normalize-url');
const ReconnectingWebSocket = require('reconnecting-websocket');

// register Monaco languages
monaco.languages.register({
    id: 'typescript',
    extensions: ['.ts'],
    aliases: ['TypeScript','ts','TS','Typescript','typescript']
})

// create Monaco editor
const value = `
let message:string = 'foo'
`;

const editor = monaco.editor.create(document.getElementById("container")!, {
    model: monaco.editor.createModel(value, 'typescript', monaco.Uri.parse('file:///C:\\Users\\foo\\Desktop\\project\\demo\\ts\\file.ts')),
    glyphMargin: true,
    theme: "vs-dark",
    lightbulb: {
        enabled: true
    }
});

// install Monaco language client services
MonacoServices.install(editor, {rootUri: "file:///C:\\Users\\foo\\Desktop\\project\\demo\\ts"});

// create the web socket
const url = createUrl('ws://localhost:3000/ws')
const webSocket = createWebSocket(url);
// listen when the web socket is opened
listen({
    webSocket,
    onConnection: connection => {
        // create and start the language client
        const languageClient = createLanguageClient(connection);
        const disposable = languageClient.start();
        connection.onClose(() => disposable.dispose());
    }
});

function createLanguageClient(connection: MessageConnection): MonacoLanguageClient {
    return new MonacoLanguageClient({
        name: "Sample Language Client",
        clientOptions: {
            // use a language id as a document selector
            documentSelector: ['typescript'],
            // disable the default error handler
            errorHandler: {
                error: () => ErrorAction.Continue,
                closed: () => CloseAction.DoNotRestart
            }
        },
        // create a language client connection from the JSON RPC connection on demand
        connectionProvider: {
            get: (errorHandler, closeHandler) => {
                return Promise.resolve(createConnection(connection, errorHandler, closeHandler))
            }
        }
    });
}

function createUrl(path: string): string {
    return normalizeUrl(path);
}

function createWebSocket(url: string): WebSocket {
    const socketOptions = {
        maxReconnectionDelay: 10000,
        minReconnectionDelay: 1000,
        reconnectionDelayGrowFactor: 1.3,
        connectionTimeout: 10000,
        maxRetries: Infinity,
        debug: false
    };
    return new ReconnectingWebSocket(url, [], socketOptions);
}
```

**server.ts**

```
import * as express from "express";

const app = express();
app.use(express.static(__dirname));
app.listen(4200);
```

**Comment by CGNonofr** [comment]
Hello

It seems you are not up to date, `MonacoServices.install` doesn't take the editor as parameter anymore

**Comment by MassMessage** [comment]
> 
> 
> Hello
> 
> It seems you are not up to date, `MonacoServices.install` doesn't take the editor as parameter anymore

Hi, thanks, I did update `MonacoServices.install(editor, {rootUri: ... }) ` by `MonacoServices.install(monaco)` . But the monaco undefined error at runtime still is there

**Comment by CGNonofr** [comment]
I just tried and it seems the example work well

Does it work without any change to the repository?

**Comment by MassMessage** [comment]
> 
> 
> I just tried and it seems the example work well
> 
> Does it work without any change to the repository?

No, I changed the main.ts (replacing monaco core import by moanco editor) and modified the client.ts and serve.ts files, as it's in the original post so that it would work for typescript language not json.

**Comment by CGNonofr** [comment]
I see you updated your original post with a screenshot of the error

Again, you're not up to date, I don't know what you did but it is not the master branch code

**Comment by MassMessage** [comment]
> 
> 
> I see you updated your original post with a screenshot of the error
> 
> Again, you're not up to date, I don't know what you did but it is not the master branch code

Can you give an sample code of a up-to-date version (using monaco edito not monaco core)?

**Comment by CGNonofr** [comment]
I'm not sure to understand what you want, I won't send you the whole repo in a comment.

Can you take the master branch then replace any occurence of `monaco-editor-core` by `monaco-editor`?

**Comment by MassMessage** [comment]
> I'm not sure to understand what you want, I won't send you the whole repo in a comment.
I want to find out in my code the part that you said is not update

> Can you take the master branch then replace any occurence of monaco-editor-core by monaco-editor?
in the example folder as for the files main.cs, client.ts, serve.ts yeah I think so. I did that it's fine until I got this runtime error. I've see other people do that but they didn't get any runtime error

**Comment by CGNonofr** [comment]
You can just compare you screenshot (
https://user-images.githubusercontent.com/11813549/130501819-fac7b382-bb78-4217-bb8c-6936c3e600b0.png) with https://github.com/TypeFox/monaco-languageclient/blob/master/client/src/monaco-workspace.ts and you'll see the differences

**Comment by MassMessage** [comment]
> 
> 
> You can just compare your screenshot (
> https://user-images.githubusercontent.com/11813549/130501819-fac7b382-bb78-4217-bb8c-6936c3e600b0.png) with https://github.com/TypeFox/monaco-languageclient/blob/master/client/src/monaco-workspace.ts and you'll see the differences

Now I see what you meant. I have no idea why they differ, I didn't change this one myself. I just cloned the repo and went to modify the previously mentioned in the example folder.

**Comment by MassMessage** [comment]
> 
> 
> You can just compare your screenshot (
> https://user-images.githubusercontent.com/11813549/130501819-fac7b382-bb78-4217-bb8c-6936c3e600b0.png) with https://github.com/TypeFox/monaco-languageclient/blob/master/client/src/monaco-workspace.ts and you'll see the differences

found why it's different: my version (in the screenshot) comes from the monaco language client's source code, it's located at monaco-languageclient\example\node_modules\monaco-languageclient\src\monaco-workspace.ts which differs from the one in the master branch. Why I don't know why since I'm using the most recent monaco languagentclient version: 0.13.0

**Comment by MassMessage** [comment]
> 
> 
> > You can just compare your screenshot (
> > https://user-images.githubusercontent.com/11813549/130501819-fac7b382-bb78-4217-bb8c-6936c3e600b0.png) with https://github.com/TypeFox/monaco-languageclient/blob/master/client/src/monaco-workspace.ts and you'll see the differences
> 
> found why it's different: my version (in the screenshot) comes from the monaco language client's source code, it's located at monaco-languageclient\example\node_modules\monaco-languageclient\src\monaco-workspace.ts which differs from the one in the master branch. Why I don't know why since I'm using the most recent monaco languagentclient version: 0.13.0

fixed this error by updating to version monaco-languageclient version 0.13.1-next.9 and changing those imports in **client.ts**:

```
import { MessageConnection } from 'vscode-jsonrpc';
import { listen } from '@codingame/monaco-jsonrpc';
import * as monaco from 'monaco-editor'
```

the monaco undefined error is gone. I got other error when connecting to the server but I think I should open another issue.

**Comment by CGNonofr** [comment]
btw, even the 0.13.1-next.9 is not up to date, I don't have the rights on the monaco-languageclient npm repo so the last version is published to @codingame/monaco-languageclient

**Comment by MassMessage** [comment]
> 
> 
> btw, even the 0.13.1-next.9 is not up to date, I don't have the rights on the monaco-languageclient npm repo so the last version is published to @codingame/monaco-languageclient

I didn't about the module @codingame/monaco-languageclient, I have tried to use here but it fail to find the module Error:  'monaco-languageclient/lib/vscode-compatibility' what should I use instead of to make it work?

**Comment by MassMessage** [comment]
> 
> 
> btw, even the 0.13.1-next.9 is not up to date, I don't have the rights on the monaco-languageclient npm repo so the last version is published to @codingame/monaco-languageclient

I didn't know about this module, are you the maintainer of it? in order to use a updated version should I rely on this on?

**Comment by LahlouY** [comment]
@MassMessage How did you solved the problem ? i keep getting some errors after using yarn run build , and when i open the browser and go to localhost/4200 i get "cannot GET /"
![image](https://user-images.githubusercontent.com/65794671/142765214-e5424639-db2d-464c-a5e0-a311a73e940b.png)

**Comment by LahlouY** [comment]
@MassMessage How did you solved the problem ? i Have the same issue, autocompletion doesn't work for me
